src/index/Index.cpp

```diff
@@ -104,9 +105,13 @@ void Index::createFromFile(const string& filename, bool allPermutations) {
     AD_CHECK(false);
   }
   // also perform unique for first permutation
-
   createPermutationPair<IndexMetaDataHmap>(&idTriples, Permutation::Pso,
                                            Permutation::Pos, true);
+  if (_addedPredicates) {
+    ExtVec stats = computeAddedPredicates(idTriples);
```

Comment:
That call breaks if it is done before writeConfigurationFile is called.
Does this function need a specific ordering of the idTriples? If not, I would suggest moving it down to the end of this function so that the writeConfigurationFile stays very late.